bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/DebugModeHandler.java

```diff
@@ -174,6 +182,22 @@ public String calculateDebugPort(Project project, String inputParms) throws Exce
     public void startDebugAttacher(Project project, String debugPort) {
         String projectName = project.getIProject().getName();
 
+        // Find if the project specifies a WLP_DEBUG_ADDRESS entry in the server.env (src/main/liberty/config).
+        // If it is specified, the port value under that variable needs to be ignored. 
+        // This is a safe place to do this because dev mode has not been started yet.
+        String port = null;
+        try {
+            Path projServerEnvPath = Paths.get(project.getPath(), "src", "main", "liberty", "config", "server.env");
+            port = readDebugPortFromServerEnv(projServerEnvPath.toFile());
```

Comment:
What's the use case here?  Why do we need to do this?

Reply:
This change polices (for default usage) the fact that we do not support project defined debug ports through a server.env (src/main/liberty/config). This allows us to more reliably connect the debugger to the JVM. 
Under certain conditions or timing events, the code added fixes issues such as: An unexpected port (project provided)  from being displayed in messages in case of timeout conditions. It Fixes the fact that under certain timing condition events, the unexpected port could be used.

Reply:
Ah, I see..thanks.

What if you happened to have within your src server.env the same port that you specified in -DdebugPort as a custom parm?

Would filtering it out like this mess up that case?

If so, would it be better to just keep track of the fact that you saw this entry in src server.env so that when you go and look at target server.env later you know to ignore/discount it?

Reply:
Then it does not matter, we continue to use -DdebugPort .